### PR TITLE
Add functional tests for configured endpoint url

### PR DESCRIPTION
**Pull request by kdaily**
*Issue #, if available:*

NA

*Description of changes:*

Add functional tests for configured endpoint URL.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: tests/functional/data/configured-endpoint-urls/test-cases/profile-tests.json
Comment (kyleknap) [inline tests/functional/data/configured-endpoint-urls/test-cases/profile-tests.json:1]: Let's also make sure there is a `feature` changelog as part of this PR so the CLI will have a changelog for configurable endpoints when it is released

File: tests/functional/test_configured_endpoint_url.py
Comment (kyleknap) [inline tests/functional/test_configured_endpoint_url.py:111]: Instead of using `BaseAWSCommandParamsTest`, let's try using the `CLIRunner` abstraction: https://github.com/aws/aws-cli/blob/86e86a5392f9084c2accbec18b228b15b14a5b84/tests/__init__.py#L39 It is more heavily used in the v2 codebase if you want examples on how to use it but the nice part about using it is that we will be able to parameterize these cases as it is not dependent on `unittest` and we can better guarantee we are not carrying over state between cases.

**Review by kyleknap — COMMENTED** [review]
Cool. Just had a few comments on this. Once we get the botocore one merged it would be also good to port over any applicable feedback from that PR to this one.

File: tests/functional/configured_endpoint_urls/test_configured_endpoint_url.py
Comment (kdaily) [inline tests/functional/configured_endpoint_urls/test_configured_endpoint_url.py:103]: long line, need to reformat

File: tests/functional/configured_endpoint_urls/test_configured_endpoint_url.py
Comment (kdaily) [inline tests/functional/configured_endpoint_urls/test_configured_endpoint_url.py:141]: This feels like its not correct, but I think I'd need to copy boilerplate to set the `AWS_DATA_PATH`. Instantiating a `CLIRunner` like this didn't work:

```
cli_runner = CLIRunner(session_stubber=session_stubber, env={})
```

File: tests/functional/configured_endpoint_urls/test_configured_endpoint_url.py
Comment (kdaily) [inline tests/functional/configured_endpoint_urls/test_configured_endpoint_url.py:143]: Remove this line.

File: tests/functional/configured_endpoint_urls/test_configured_endpoint_url.py
Comment (kdaily) [inline tests/functional/configured_endpoint_urls/test_configured_endpoint_url.py:127]: I think I can move the creation and config of the runner into a helper fixture, will work on that. Wanted to be sure this was functional!

**Review by kyleknap — COMMENTED** [review]
Looks fine. Just had a few more comments to go along with the comments you left yourself

**Review by kyleknap — COMMENTED** [review]
This looks good. I'll approve it once we switch it out of draft and get a clean run of CI once the botocore PR is merged.

**Review by kyleknap — COMMENTED** [review]
I think we need to rebase this PR one more time off of develop. Looking at the CI logs, I'm seeing the botocore being installed from it's `develop` branch is being clobbered by the one from PyPI since this PR currently requires botocore 1.30.0:
```
Installing collected packages: PyYAML, pyasn1, docutils, rsa, botocore, awscli
  Attempting uninstall: botocore
    Found existing installation: botocore 1.30.1
    Uninstalling botocore-1.30.1:
      Successfully uninstalled botocore-1.30.1
Successfully installed PyYAML-5.4.1 awscli-1.28.0 botocore-1.30.0 docutils-0.16 pyasn1-0.5.0 rsa-4.7.2
```

**Review by kyleknap — APPROVED** [review]
Looks good :ship:

**Pull request merged**
